mllib/src/test/scala/org/apache/spark/mllib/linalg/BLASSuite.scala

```diff
@@ -257,32 +257,60 @@ class BLASSuite extends FunSuite {
       new DenseMatrix(4, 3, Array(0.0, 1.0, 0.0, 0.0, 2.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 3.0))
     val sA = new SparseMatrix(4, 3, Array(0, 1, 3, 4), Array(1, 0, 2, 3), Array(1.0, 2.0, 1.0, 3.0))
 
-    val x = new DenseVector(Array(1.0, 2.0, 3.0))
+    val dx = new DenseVector(Array(1.0, 2.0, 3.0))
+    val sx = dx.toSparse
     val expected = new DenseVector(Array(4.0, 1.0, 2.0, 9.0))
 
-    assert(dA.multiply(x) ~== expected absTol 1e-15)
-    assert(sA.multiply(x) ~== expected absTol 1e-15)
+    assert(dA.multiply(dx) ~== expected absTol 1e-15)
+    assert(sA.multiply(dx) ~== expected absTol 1e-15)
 
```

Comment:
`Matrix.multiply` has a problem. Now its signature is `multiply(y: DenseVector)`. Turning it to `multiply(y: Vector)` or adding new `multiply(y: Vector)` can't pass binary compatibility check as I did in #6189. I am asking @mengxr whether we can add it into `MimaExcludes`.